### ISSUE TITLE
Add support for Vyper decimal10 ABI type

### ISSUE THREAD

**Issue by darcius** [ethereum/web3.js#1522]
## What was wrong?

There is currently no support for Vyper abi type decimal10. Although Vyper is in alpha, it is being increasingly used, most recently and importantly in the Casper contract. If you currently attempt to deploy/read a contract with this ABI type, you get `Error: Invalid solidity type: decimal10`

## How can it be fixed?

Add support for decimal10 abi type :) web3.py are also working on it https://github.com/ethereum/web3.py/issues/700

**Comment by djrtwo** [comment]
Looks like vyper might move to just supporting `fixedMxN` https://github.com/ethereum/vyper/issues/566

If so, Casper will be migrating to `fixed168x10` rather than `decimal10`

**Comment by darcius** [comment]
Thanks Danny, suspected that might be the case. We couldn't wait though :) So we added support for Decimal10 ourselves, have been holding off submitting a PR here until the direction is finalised for it. I've only tested this with Casper so far and it works well atm, for anyone else who wants Vyper Decimal10 support in Web3js in the interim, feel free to clone our branch here.

https://github.com/rocket-pool/web3.js/blob/decimal10-support/packages/web3-eth-abi/src/types/decimal.js
